Marlin/src/module/tool_change.cpp

```diff
@@ -765,7 +765,11 @@ void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool n
 
       set_destination_from_current();
 
-      if (!no_move) {
+      if (!no_move
+        #if ENABLED(DUAL_X_CARRIAGE)
+          && dual_x_carriage_mode != DXC_FULL_CONTROL_MODE
+        #endif
+      ) {
```

Comment:
For `DUAL_X_CARRIAGE` this block will only be entered when `dual_x_carriage_mode != DXC_FULL_CONTROL_MODE`….

Reply:
This is correct. That section blocks the Z raise. Full control mode only needs to apply hotend offsets and change active steppers. This only encompasses lines 763 - 778 from my original snapshot,